File: Daily Challanges/2026-07-July/2026-07-26.js
function letterDistance(s1, s2) {
  let total = 0;

  for (let i = 0; i < s1.length; i++) {
    const diff = Math.abs(s1.charCodeAt(i) - s2.charCodeAt(i));
    total += Math.min(diff, 26 - diff);
  }

  return total;
}

console.log(letterDistance("abc", "bcd"));
console.log(letterDistance("abc", "xyz"));
console.log(letterDistance("encrypt", "decrypt"));
console.log(letterDistance("algorithm", "codeblock"));
console.log(letterDistance("lobster", "penguin"));
console.log(letterDistance("alligator", "crocodile"));

File: Daily Challanges/2026-07-July/2026-07-26.py
def letter_distance(s1, s2):
    total = 0

    for i in range(len(s1)):
        diff = abs(ord(s1[i]) - ord(s2[i]))
        total += min(diff, 26 - diff)

    return total


print(letter_distance("abc", "bcd"))
print(letter_distance("abc", "xyz"))
print(letter_distance("encrypt", "decrypt"))
print(letter_distance("algorithm", "codeblock"))
print(letter_distance("lobster", "penguin"))
print(letter_distance("alligator", "crocodile"))